kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/AdapterGenerator.kt

```diff
@@ -133,9 +133,16 @@ internal class AdapterGenerator(
     }
 
     result.addProperty(optionsProperty)
+
+    val delegateKeyCount = propertyList.groupingBy { it.delegateKey }.eachCount()
+
     for (uniqueAdapter in propertyList.distinctBy { it.delegateKey }) {
```

Comment:
since you’re doing groupingBy and distinctBy in two adjacent calls, could you only do groupingBy ? As-is it’s doing 2x as much work and it’s more awkward to follow